modules/apps/29-fee/keeper/msg_server.go

```diff
@@ -1,4 +1,37 @@
 package keeper
 
-// TODO
-//var _ types.MsgServer = Keeper{}
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+)
+
+var _ types.MsgServer = Keeper{}
+
+// RegisterCounterpartyAddress is called by the relayer on each channelEnd and allows them to specify their counterparty address before relaying
+// This ensures they will be properly compensated for forward relaying since destination chain must send back relayer's source address (counterparty address) in acknowledgement
+// This function may be called more than once by relayer, in which case, latest counterparty address is always used.
+func (k Keeper) RegisterCounterpartyAddress(goCtx context.Context, msg *types.MsgRegisterCounterpartyAddress) (*types.MsgRegisterCounterpartyAddressResponse, error) {
```

Comment:
@AdityaSripal I'm wondering how we are planning to verify that the relayer using this endpoint will only be able to set a `CounterpartyAddress` for themselves, and not change other relayer mappings.